tinygrad/renderer/cstyle.py

```diff
@@ -222,14 +222,32 @@ class OpenCLRenderer(CStyleLanguage):
   float4 = "(float4)"
   code_for_workitem = {"g": lambda x: f"get_group_id({x})", "l": lambda x: f"get_local_id({x})", "i": lambda x: f"get_global_id({x})"}
   uses_vload = True
-  type_map = { dtypes.uint8: "uchar", dtypes.uint32: "uint", dtypes.uint16: "ushort", dtypes.uint64: "ulong" }
+  type_map = { dtypes.uint8: "uchar", dtypes.uint32: "uint", dtypes.uint16: "ushort", dtypes.uint64: "ulong", dtypes.bfloat16: "ushort" }
   def render_cast(self, x, var_dtype, bitcast=False) -> str:
     return f"as_{self.render_dtype(var_dtype)}({x})" if bitcast else super().render_cast(x, var_dtype)
 
   def render_kernel(self, function_name, kernel, bufs, uops, prefix=None) -> str:
-    if any(uop.dtype == dtypes.half for uop in uops): prefix = ["#pragma OPENCL EXTENSION cl_khr_fp16 : enable"]
+    if any(uop.dtype == dtypes.half for uop in uops): prefix = (lambda p: p + prefix if prefix else p)(["#pragma OPENCL EXTENSION cl_khr_fp16 : enable"]) # noqa: E501
     return super().render_kernel(function_name, kernel, bufs, uops, prefix)
 
+class IntelRenderer(OpenCLRenderer):
+  device, kernel_prefix = "INTEL", "__attribute__((intel_reqd_sub_group_size(8)))\n" + "__kernel "
```

Comment:
In device.py, I added an array of specialized cl devices, so it behaves just like GPU.

Reply:
can we keep `GPU` and use `suffix` (like ptx)?

Reply:
Main issue is that I need to be able to check the device of the WMMA uop in uops.py so that flops get counted correctly. Maybe I'm misunderstanding the intended semantics, but unlike cuda/ptx, which are intended to be run on the same hardware, INTEL is meant to be used with intel gpus specifically, so it seems more obvious than using the suffix.